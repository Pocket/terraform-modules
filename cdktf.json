{
  "language": "typescript",
  "app": "npm run --silent compile && node dist/src/example.js",
  "terraformProviders": [
    "pagerduty/pagerduty@~> 1.8.0",
    "archive@~> 1.3.0",
    "local@~> 2.1.0"
  ]
}
