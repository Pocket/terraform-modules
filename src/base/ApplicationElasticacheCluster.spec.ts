import { TerraformStack, Testing } from 'cdktf';
import {
  ApplicationElasticacheCluster,
  ApplicationElasticacheClusterProps,
  ApplicationElasticacheEngine,
} from './ApplicationElasticacheCluster';

const BASE_CONFIG: ApplicationElasticacheClusterProps = {
  allowedIngressSecurityGroupIds: [],
  engine: ApplicationElasticacheEngine.MEMCACHED,
  subnetIds: ['1234-123'],
  vpcId: 'cool-vpc',
  prefix: `abides-dev`,
};

test('renders memcached with minimal config', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  new ApplicationElasticacheCluster(stack, 'testMemcached', BASE_CONFIG);

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders memcached with tags', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  const config: ApplicationElasticacheClusterProps = {
    ...BASE_CONFIG,
    tags: {
      letsgo: 'bowling',
      donnie: 'throwinrockstonight',
    },
  };
  new ApplicationElasticacheCluster(stack, 'testMemcached', config);

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders memcached with node change', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');

  const config: ApplicationElasticacheClusterProps = {
    ...BASE_CONFIG,
    node: {
      size: 'cache.m4.2xlarge',
      count: 5,
    },
  };
  new ApplicationElasticacheCluster(stack, 'testMemcached', config);

  expect(Testing.synth(stack)).toMatchSnapshot();
});

test('renders redis with minimal config', () => {
  const app = Testing.app();
  const stack = new TerraformStack(app, 'test');
  const config: ApplicationElasticacheClusterProps = {
    ...BASE_CONFIG,
    engine: ApplicationElasticacheEngine.REDIS,
  };
  new ApplicationElasticacheCluster(stack, 'testRedis', config);

  expect(Testing.synth(stack)).toMatchSnapshot();
});
