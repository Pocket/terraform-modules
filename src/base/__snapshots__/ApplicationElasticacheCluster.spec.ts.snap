// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders memcached with minimal config 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"data\\": {
    \\"aws_vpc\\": {
      \\"testMemcached_vpc_192C8AF1\\": {
        \\"filter\\": [
          {
            \\"name\\": \\"vpc-id\\",
            \\"values\\": [
              \\"cool-vpc\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/vpc\\",
            \\"uniqueId\\": \\"testMemcached_vpc_192C8AF1\\"
          }
        }
      }
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"testMemcached_elasticache_security_group_F23969B6\\": {
        \\"description\\": \\"Managed by Terraform\\",
        \\"ingress\\": [
          {
            \\"cidr_blocks\\": null,
            \\"description\\": null,
            \\"from_port\\": 11211,
            \\"ipv6_cidr_blocks\\": null,
            \\"prefix_list_ids\\": null,
            \\"protocol\\": \\"tcp\\",
            \\"security_groups\\": [],
            \\"self\\": null,
            \\"to_port\\": 11211
          }
        ],
        \\"name_prefix\\": \\"abides-dev\\",
        \\"vpc_id\\": \\"\${data.aws_vpc.testMemcached_vpc_192C8AF1.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_security_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_security_group_F23969B6\\"
          }
        }
      }
    },
    \\"aws_elasticache_subnet_group\\": {
      \\"testMemcached_elasticache_subnet_group_609E6233\\": {
        \\"name\\": \\"abides-dev-ElasticacheSubnetGroup\\",
        \\"subnet_ids\\": [
          \\"1234-123\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_subnet_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_subnet_group_609E6233\\"
          }
        }
      }
    },
    \\"aws_elasticache_cluster\\": {
      \\"testMemcached_elasticache_cluster_4853BD62\\": {
        \\"apply_immediately\\": true,
        \\"cluster_id\\": \\"abides-dev\\",
        \\"engine\\": \\"memcached\\",
        \\"engine_version\\": \\"1.6.6\\",
        \\"node_type\\": \\"cache.t2.micro\\",
        \\"num_cache_nodes\\": 2,
        \\"parameter_group_name\\": \\"default.memcached1.6\\",
        \\"port\\": 11211,
        \\"security_group_ids\\": [
          \\"\${aws_security_group.testMemcached_elasticache_security_group_F23969B6.id}\\"
        ],
        \\"subnet_group_name\\": \\"\${aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233.name}\\",
        \\"depends_on\\": [
          \\"aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233\\",
          \\"aws_security_group.testMemcached_elasticache_security_group_F23969B6\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_cluster\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_cluster_4853BD62\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders memcached with node change 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"data\\": {
    \\"aws_vpc\\": {
      \\"testMemcached_vpc_192C8AF1\\": {
        \\"filter\\": [
          {
            \\"name\\": \\"vpc-id\\",
            \\"values\\": [
              \\"cool-vpc\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/vpc\\",
            \\"uniqueId\\": \\"testMemcached_vpc_192C8AF1\\"
          }
        }
      }
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"testMemcached_elasticache_security_group_F23969B6\\": {
        \\"description\\": \\"Managed by Terraform\\",
        \\"ingress\\": [
          {
            \\"cidr_blocks\\": null,
            \\"description\\": null,
            \\"from_port\\": 11211,
            \\"ipv6_cidr_blocks\\": null,
            \\"prefix_list_ids\\": null,
            \\"protocol\\": \\"tcp\\",
            \\"security_groups\\": [],
            \\"self\\": null,
            \\"to_port\\": 11211
          }
        ],
        \\"name_prefix\\": \\"abides-dev\\",
        \\"vpc_id\\": \\"\${data.aws_vpc.testMemcached_vpc_192C8AF1.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_security_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_security_group_F23969B6\\"
          }
        }
      }
    },
    \\"aws_elasticache_subnet_group\\": {
      \\"testMemcached_elasticache_subnet_group_609E6233\\": {
        \\"name\\": \\"abides-dev-ElasticacheSubnetGroup\\",
        \\"subnet_ids\\": [
          \\"1234-123\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_subnet_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_subnet_group_609E6233\\"
          }
        }
      }
    },
    \\"aws_elasticache_cluster\\": {
      \\"testMemcached_elasticache_cluster_4853BD62\\": {
        \\"apply_immediately\\": true,
        \\"cluster_id\\": \\"abides-dev\\",
        \\"engine\\": \\"memcached\\",
        \\"engine_version\\": \\"1.6.6\\",
        \\"node_type\\": \\"cache.m4.2xlarge\\",
        \\"num_cache_nodes\\": 5,
        \\"parameter_group_name\\": \\"default.memcached1.6\\",
        \\"port\\": 11211,
        \\"security_group_ids\\": [
          \\"\${aws_security_group.testMemcached_elasticache_security_group_F23969B6.id}\\"
        ],
        \\"subnet_group_name\\": \\"\${aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233.name}\\",
        \\"depends_on\\": [
          \\"aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233\\",
          \\"aws_security_group.testMemcached_elasticache_security_group_F23969B6\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_cluster\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_cluster_4853BD62\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders memcached with tags 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"data\\": {
    \\"aws_vpc\\": {
      \\"testMemcached_vpc_192C8AF1\\": {
        \\"filter\\": [
          {
            \\"name\\": \\"vpc-id\\",
            \\"values\\": [
              \\"cool-vpc\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/vpc\\",
            \\"uniqueId\\": \\"testMemcached_vpc_192C8AF1\\"
          }
        }
      }
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"testMemcached_elasticache_security_group_F23969B6\\": {
        \\"description\\": \\"Managed by Terraform\\",
        \\"ingress\\": [
          {
            \\"cidr_blocks\\": null,
            \\"description\\": null,
            \\"from_port\\": 11211,
            \\"ipv6_cidr_blocks\\": null,
            \\"prefix_list_ids\\": null,
            \\"protocol\\": \\"tcp\\",
            \\"security_groups\\": [],
            \\"self\\": null,
            \\"to_port\\": 11211
          }
        ],
        \\"name_prefix\\": \\"abides-dev\\",
        \\"tags\\": {
          \\"letsgo\\": \\"bowling\\",
          \\"donnie\\": \\"throwinrockstonight\\"
        },
        \\"vpc_id\\": \\"\${data.aws_vpc.testMemcached_vpc_192C8AF1.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_security_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_security_group_F23969B6\\"
          }
        }
      }
    },
    \\"aws_elasticache_subnet_group\\": {
      \\"testMemcached_elasticache_subnet_group_609E6233\\": {
        \\"name\\": \\"abides-dev-ElasticacheSubnetGroup\\",
        \\"subnet_ids\\": [
          \\"1234-123\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_subnet_group\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_subnet_group_609E6233\\"
          }
        }
      }
    },
    \\"aws_elasticache_cluster\\": {
      \\"testMemcached_elasticache_cluster_4853BD62\\": {
        \\"apply_immediately\\": true,
        \\"cluster_id\\": \\"abides-dev\\",
        \\"engine\\": \\"memcached\\",
        \\"engine_version\\": \\"1.6.6\\",
        \\"node_type\\": \\"cache.t2.micro\\",
        \\"num_cache_nodes\\": 2,
        \\"parameter_group_name\\": \\"default.memcached1.6\\",
        \\"port\\": 11211,
        \\"security_group_ids\\": [
          \\"\${aws_security_group.testMemcached_elasticache_security_group_F23969B6.id}\\"
        ],
        \\"subnet_group_name\\": \\"\${aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233.name}\\",
        \\"tags\\": {
          \\"letsgo\\": \\"bowling\\",
          \\"donnie\\": \\"throwinrockstonight\\"
        },
        \\"depends_on\\": [
          \\"aws_elasticache_subnet_group.testMemcached_elasticache_subnet_group_609E6233\\",
          \\"aws_security_group.testMemcached_elasticache_security_group_F23969B6\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testMemcached/elasticache_cluster\\",
            \\"uniqueId\\": \\"testMemcached_elasticache_cluster_4853BD62\\"
          }
        }
      }
    }
  }
}"
`;

exports[`renders redis with minimal config 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"test\\"
    }
  },
  \\"data\\": {
    \\"aws_vpc\\": {
      \\"testRedis_vpc_442CAD26\\": {
        \\"filter\\": [
          {
            \\"name\\": \\"vpc-id\\",
            \\"values\\": [
              \\"cool-vpc\\"
            ]
          }
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testRedis/vpc\\",
            \\"uniqueId\\": \\"testRedis_vpc_442CAD26\\"
          }
        }
      }
    }
  },
  \\"resource\\": {
    \\"aws_security_group\\": {
      \\"testRedis_elasticache_security_group_36C6CB6D\\": {
        \\"description\\": \\"Managed by Terraform\\",
        \\"ingress\\": [
          {
            \\"cidr_blocks\\": null,
            \\"description\\": null,
            \\"from_port\\": 6379,
            \\"ipv6_cidr_blocks\\": null,
            \\"prefix_list_ids\\": null,
            \\"protocol\\": \\"tcp\\",
            \\"security_groups\\": [],
            \\"self\\": null,
            \\"to_port\\": 6379
          }
        ],
        \\"name_prefix\\": \\"abides-dev\\",
        \\"vpc_id\\": \\"\${data.aws_vpc.testRedis_vpc_442CAD26.id}\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testRedis/elasticache_security_group\\",
            \\"uniqueId\\": \\"testRedis_elasticache_security_group_36C6CB6D\\"
          }
        }
      }
    },
    \\"aws_elasticache_subnet_group\\": {
      \\"testRedis_elasticache_subnet_group_389F6AE7\\": {
        \\"name\\": \\"abides-dev-ElasticacheSubnetGroup\\",
        \\"subnet_ids\\": [
          \\"1234-123\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testRedis/elasticache_subnet_group\\",
            \\"uniqueId\\": \\"testRedis_elasticache_subnet_group_389F6AE7\\"
          }
        }
      }
    },
    \\"aws_elasticache_cluster\\": {
      \\"testRedis_elasticache_cluster_014A9AA6\\": {
        \\"apply_immediately\\": true,
        \\"cluster_id\\": \\"abides-dev\\",
        \\"engine\\": \\"redis\\",
        \\"engine_version\\": \\"6.x\\",
        \\"node_type\\": \\"cache.t2.micro\\",
        \\"num_cache_nodes\\": 2,
        \\"parameter_group_name\\": \\"default.redis6.x\\",
        \\"port\\": 6379,
        \\"security_group_ids\\": [
          \\"\${aws_security_group.testRedis_elasticache_security_group_36C6CB6D.id}\\"
        ],
        \\"subnet_group_name\\": \\"\${aws_elasticache_subnet_group.testRedis_elasticache_subnet_group_389F6AE7.name}\\",
        \\"depends_on\\": [
          \\"aws_elasticache_subnet_group.testRedis_elasticache_subnet_group_389F6AE7\\",
          \\"aws_security_group.testRedis_elasticache_security_group_36C6CB6D\\"
        ],
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"test/testRedis/elasticache_cluster\\",
            \\"uniqueId\\": \\"testRedis_elasticache_cluster_014A9AA6\\"
          }
        }
      }
    }
  }
}"
`;
